exec/plugin-commands-script-runners/src/run.ts

```diff
@@ -158,6 +164,7 @@ export type RunOpts =
   & { recursive?: boolean }
   & Pick<Config,
   | 'bin'
+  | 'checkDepsBeforeRunScripts'
```

Comment:
I’d suggest a shorter setting name
```suggestion
  | ‘verifyDepsBeforeRun'
```

Also, I think the same checks should be applied to the exec commands. In that case it will make more sense to name it
```suggestion
  | 'verifyDepsBeforeExec'
```

but `run` could work for both

Reply:
Now the feature has been added to `exec`: https://github.com/pnpm/pnpm/pull/8645/commits/2da7c134c8a9c30838429d05463aeab8ec726102